src/main/java/net/earthcomputer/clientcommands/c2c/PacketCacheHelper.java

```diff
@@ -0,0 +1,24 @@
+package net.earthcomputer.clientcommands.c2c;
+
+import com.google.common.cache.CacheBuilder;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Set;
+
+public class PacketCacheHelper {
```

Comment:
Not keen on the name of this class tbh, although I don't immediately have any better suggestions. It's not indicative of its functionality.

Reply:
What about `ExpiringPacketCache`?

Reply:
It's more like a class to prevent you getting your own messages, so maybe the name should reflect that?

Reply:
`OutgoingPacketCache`, `OutgoingPacketFilter`, ...?